packages/client/src/runtime/core/engines/common/Engine.ts

```diff
@@ -147,6 +147,14 @@ export interface EngineConfig {
    * in the current working directory. This usually means it has been bundled.
    */
   isBundled?: boolean
+
+  /**
+   * Loads the raw wasm module for the wasm query engine. This configuration is
+   * generated specifically for each type of client, eg. Node.js client and Edge
+   * clients will have different implementations.
+   * @remarks this is a callback on purpose, we only load the wasm if needed.
+   */
+  getQueryEngineWasmModule?: () => Promise<any>
```

Comment:
```suggestion
  getQueryEngineWasmModule?: () => Promise<Library>
```

From `packages/client/src/runtime/core/engines/library/types/Library.ts`

Reply:
It's not exactly working this way, at this stage it is just a raw wasm module, the instance is later created here https://github.com/prisma/prisma/blob/cfd3296895a41a091c46db08ac501be44cdec74a/packages/client/src/runtime/core/engines/library/WasmLibraryLoader.ts/#L14-L16